### PR TITLE
Implement MultiDB Support and Tests

### PR DESCRIPTION
**Pull request by tom-andersen**

### CODE REVIEW

**Comment by mortenbekditlevsen** [comment]
Wow! This is a dream come true! 🥰

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 1]
```diff
@@ -43,22 +43,22 @@ NS_SWIFT_NAME(Firestore)
...
  * invocation returns the same `Firestore` object.
  *
- * @return The `Firestore` instance.
+ * @return The default `Firestore` instance.
  */
 + (instancetype)firestore NS_SWIFT_NAME(firestore());
 
 /**
- * Creates, caches, and returns a `Firestore` object for the specified _app_. Each subsequent
+ * Creates, caches, and returns default `Firestore` object for the specified _app_. Each subsequent
```
Comment (wu-hui) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:54]: _app_ -> app

File: Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm [hunk 2]
```diff
@@ -1744,4 +1753,37 @@ - (void)testWaitForPendingWritesCompletesWhenOfflineIfNoPending {
...
+- (void)testSameNamedDbIsSame {
+  [FIRApp configure];
+  FIRApp *app = [FIRApp defaultApp];
+  FIRFirestore *db1 = [FIRFirestore firestoreForApp:app database:@"myDb"];
+  FIRFirestore *db2 = [FIRFirestore firestoreForDatabase:@"myDb"];
+
+  XCTAssertIdentical(db1, db2);
+}
+
+- (void)testNamedDbHaveDifferentPersistence {
```
Comment (wu-hui) [inline Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm:1778]: FWIW, here it is testing if they are the same instance, not the same persistence underlying..maybe this is a typo?

Comment (wu-hui) [reply Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm]: Persistence -> Instance ?

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h (hunk 1 above)
Comment (tom-andersen) [reply Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h]: I think the underscores are required to document linking. It follows the pattern from elsewhere.

**Pull request merged**